### DISCUSSION TITLE
Passing Additional Parameter in SignIn()

### DISCUSSION THREAD

**Discussion by Saidul-Shiblee** [nextauthjs/next-auth#6854] [Help]
Is it possible to pass additional parameters to the /authorize endpoint through the third argument of signIn() while the provider is "email"?

**Comment by nhuethmayr** [comment]
Had the same problem and it seems there is not an easy way to achieve this without writing your own provider. I ended up working around it by adding 2 email providers and distinguishing them by name.

1) Add 2 providers
![image](https://user-images.githubusercontent.com/26460692/222441871-c1889e9a-e631-472e-a191-f0cfcd46c33f.png)

2) Distinguish inside `signIn`
![image](https://user-images.githubusercontent.com/26460692/222442144-51a9432e-e350-4fdf-85e4-b8d15b7c0a32.png)

3) Calling different providers depending on use-case
![image](https://user-images.githubusercontent.com/26460692/222442271-11125969-3b32-4391-a89a-17a350f7e857.png)

This is not elegant at all but it did work for me
